### ISSUE TITLE
SGE setting -V breaks setting metadata on the cluster

### ISSUE THREAD

**Issue by peterjc** [galaxyproject/galaxy#7403]
I've seen this error reported every so often on the mailing list, but couldn't find an issue on it:

> An error occurred setting the metadata for this dataset
> Set it manually or retry auto-detection

Apparently a standard work around is to change  ``retry_metadata_internally=true`` to ``retry_metadata_internally=false`` in your configuration.

We just saw this error specific to one user account (which was an important clue as our Galaxy submits jobs to an SGE cluster where they run as the user) on Trimmomatic and FastQC jobs. This in turn broke workflows which would have used the green outputs which were lacking metadata.

We could fix this by removing the ``-V`` setting in the user's ``~/sge_request`` file.

i.e. When submitting the compute job, Galaxy must be defensively avoiding the user's default, but when submitting the metadata job, it was not - and for this user their settings broke this.

This was on an old server running ``release_16.01``, and unfortunately I am not in a position to try this on the latest release.

**Comment by bernt-matthias** [comment]
I have a question to your question: What do you mean with "submitting the metadata job"?

**Comment by peterjc** [comment]
Galaxy processes each dataset to determine metadata like the file format (if not specified), peep text, etc. This sometimes fails, usually on data import, or for tools with somewhat dynamic output (which appears to include Trimmomatic and FastQC jobs).

As I understand it, the metadata processing is usually done externally from the Galaxy server by submitting jobs to the cluster, rather than internally on the web-server running Galaxy itself.

The message comes from here (in v18.09):

https://github.com/galaxyproject/galaxy/blob/v18.09/client/galaxy/scripts/mvc/dataset/dataset-li.js#L365

The code setting that state looks to be here:

https://github.com/galaxyproject/galaxy/blob/v18.09/lib/galaxy/tools/__init__.py#L2251

That suggests function ``external_metadata_set_successfully`` fails, which is defined here:

https://github.com/galaxyproject/galaxy/blob/v18.09/lib/galaxy/model/metadata.py#L795

I suspect the problem lies in ``setup_external_metadata`` within that file which sets up the cluster job to set metadata. but where?

**Comment by mvdbeek** [comment]
> i.e. When submitting the compute job, Galaxy must be defensively avoiding the user's default, but when submitting the metadata job, it was not - and for this user their settings broke this.

Yeah, I bet this is real user specific, and that triggering the setup_external_metadata doesn't do the user switch at all. I guess 16.01 is too old to diagnose this further, but we might be able to set up tests for this.
